Remove `concurrent-ruby` dependency from gemspec; bump version to 1.0.1

patbl opened the pull request.

patbl committed:
Make `concurrent-ruby` a development (not runtime) dependency

It was added as a runtime dependency in
https://github.com/academia-edu/cached_counts/pull/17. It **is** a
runtime dependency, but only a transitive one (through Rails). The only
reason we need to specify it is that on Rails versions previous to Rails
7.1 or https://github.com/rails/rails/tree/7-0-stable (see
https://github.com/rails/rails/pull/54264 for the relevant fix), you'd
get an error with `concurrent-ruby` 1.3.5.

Once we no longer test against Rails 7.0, we'll probably be able to
remove this altogether.

patbl committed:
Bump version to 0.8.0

kevinwuhoo reviewed (approved):
![](https://media4.giphy.com/media/1wpaCOVdglMSYgjCum/giphy.gif?cid=7896f6a63dz2itxowg9m7sgg265mf6tgksaz6e8ftxs89les&ep=v1_gifs_search&rid=giphy.gif&ct=g)

patbl committed:
CI: Bump Ruby version to 3.2

patbl committed:
Remove Pry

It's past its pryme.

patbl committed:
Remove support for Rails < 7.0; test Rails 7.0+

* Test Rails 7.0, 7.1, and 8.0
* Require at least Rails 7.0 in gemspec
* Parameterize CircleCI config
* Don't use Node image in CircleCI (it's not necessary)
* Use SQLite3 gem 1.x for Rails 7.0 and 7.1 and 2.x for Rails 8.0
  (because those combinations were compatible)

patbl committed:
Bump version to 1.0.0

I'm doing this 'cause it removes support for version of Rails < 7.0.

patbl committed:
Use  `RAILS_VERSION` instead of `ACTIVERECORD_VERSION` throughout

We already require Rails as a runtime dependency, and using "Rails"
everywhere seems more straightforward and consistent.

patbl committed:
Remove development and test Gemfile groups

These seem unused.

patbl commented:
![hai_doggy](https://github.com/user-attachments/assets/8216d222-f8b4-4399-a45a-87f8010363ab)

patbl committed:
Bump version to 1.0.1

I already pushed version 1.0.0.

patbl commented:
@scpike You can review this just for correctness (as in, it won't cause any problems for the Rails app). We don't actively develop this gem, so the code doesn't have to be super nice.

scpike reviewed (approved):
Looks fine to me - but concurrent-ruby can possibly just be dropped as a dependency? We added it because of an incompatibility with concurrent-ruby 1.3.5, but that wasn't actually true. I think it's just that the gem was incompatible with having concurrent-ruby 1.3.5 installed _and also_ not having `logger` required?

Might not be worth digging any further into.

patbl commented:
@scpike The reason I left `concurrent-ruby` in there was that the tests will fail on Rails 7.0 otherwise:

```
pat@MacBookPro:~/academia/academia-edu/cached_counts patbl/remove-concurrent-ruby-dependency-from-gemspec* RAILS_VERSION=7.0.8.7 bundle exec rspec

An error occurred while loading ./spec/caches_count_of_spec.rb.
Failure/Error: require 'active_record'

NameError:
  uninitialized constant ActiveSupport::LoggerThreadSafeLevel::Logger
# ./spec/spec_helper.rb:3:in `<top (required)>'
# ./spec/caches_count_of_spec.rb:1:in `<top (required)>'

An error occurred while loading ./spec/caches_count_where_spec.rb.
Failure/Error: require 'active_record'

NameError:
  uninitialized constant ActiveSupport::LoggerThreadSafeLevel::Logger
# ./spec/spec_helper.rb:3:in `<top (required)>'
# ./spec/caches_count_where_spec.rb:1:in `<top (required)>'
No examples found.


Finished in 0.00005 seconds (files took 0.343 seconds to load)
0 examples, 0 failures, 2 errors occurred outside of examples
```

Maybe it would be better to make it conditional in the Gemfile, though, so it's more obvious why it's there and when it'll be OK to remove (when we stop testing against Rails 7.0). That seems like a quick change.

patbl committed:
Make it more obvious why `concurrent-ruby` is listed as a dependency

And also make it clearer when it'll be OK to remove it.

patbl committed:
Bump version to 1.0.2

patbl commented:
As you said, an alternative would be to require `logger`. Maybe I could just add that as a conditional somewhere instead of adding `concurrent-ruby` as a dependency.

patbl commented:
OK, I did that.
